packages/u/usysconf-epoch/files/epoch.sh

```diff
@@ -97,37 +149,66 @@ create_compat_link () {
 copy_or_hard_link () {
     local file_to_move="$1"
     local destination="$2"
-    local cp_args="$3"
-
-    local dest_parent=$($DIRNAME "$destination")
+    shift 2
+    local cp_args=("$@")
+    local dest_parent source_device dest_device
 
-    local source_device=$($STAT -c "%D" "$file_to_move")
-    local dest_device=$($STAT -c "%D" "$dest_parent")
+    dest_parent=$($DIRNAME "$destination")
+    source_device=$($STAT -c "%D" "$file_to_move")
+    dest_device=$($STAT -c "%D" "$dest_parent")
 
     if [[ "$source_device" == "$dest_device" ]]; then
-        cp_args="$cp_args --link"
+        cp_args+=("--link")
     fi
 
-    $CP $cp_args "$file_to_move" "$destination"
+    $CP "${cp_args[@]}" "$file_to_move" "$destination"
```

Comment:
`"${cp_args[@]}"` is going to result in multiple arguments to cp being wrapped in quotes, right? IE, if the args are `--foo` and `--bar` then it would run `cp "--foo --bar"` right? I imagine that would break the argument parsing which is probably not desired.

Reply:
Nope, this is how Bash expands arrays to the quoted parts:

```
$ arr=("a b" "c d")
$ for i in "${arr[@]}"; do echo $i; done
a b
c d
```

Completely sane and logical syntax :smiling_face_with_tear: